src/twisted/names/dns.py

```diff
@@ -1902,6 +1917,79 @@ def __hash__(self):
 
 
 
+@implementer(IEncodable, IRecord)
+class Record_SSHFP(tputil.FancyEqMixin, tputil.FancyStrMixin):
+    """
+    A record containing the fingerprint of an SSH key.
+
+    @type keytype: L{int}
+    @ivar keytype: The SSH key algorithm or key type.
+    Note that the numbering used for SSH key algorithms is specific
+    to the SSHFP record type, and is not the same as the nubering
+    used for KEY or SIG records.
+
+    @type fptype: L{int}
+    @ivar fptype: The fingerprint type,
+                  such as L{FPTYPE_SHA1} or L{FPTYPE_SHA256}.
+
+    @type fingerprint: L{bytes}
+    @ivar fingerprint: The key's fingerprint, e.g. a 32-byte SHA-256 digest.
+
+    @cvar KEYTYPE_RSA: The key type value for C{ssh-rsa} keys.
+    @cvar KEYTYPE_DSS: The key type value for C{ssh-dss} keys.
+    @cvar KEYTYPE_ECDSA: The key type value for C{ecdsa-sha2-*} keys.
+    @cvar KEYTYPE_Ed25519: The key type value for C{ed25519} keys.
+
+    @cvar FPTYPE_SHA1: The fptype value for SHA-1 fingerprints.
+    @cvar FPTYPE_SHA256: The fptype value for SHA-256 fingerprints.
+
+    @see: U{RFC 4255 <https://tools.ietf.org/html/rfc4255>}
+          and
+          U{RFC 6594 <https://tools.ietf.org/html/rfc6594>}
+    """
+    fancybasename = "SSHFP"
+    compareAttributes = ('keytype', 'fptype', 'fingerprint', 'ttl')
+    showAttributes = ('keytype', 'fptype', ('fingerprint', _nicebytes))
+    TYPE = SSHFP
+
+    KEYTYPE_RSA = 1
+    KEYTYPE_DSS = 2
+    KEYTYPE_ECDSA = 3
+    KEYTYPE_Ed25519 = 4
+    _KEYTYPES = [
```

Comment:
is this mapping used somewhere?

Reply:
I was going to use it to produce a "nice" `repr` but decided not to right now. Deleted the mapping.

Reply:
yes... I used to do that... but I don't think that it pays to do that only for debugging.